pkg/syncserver/sync_server.go

```diff
@@ -265,6 +290,22 @@ func (s *Server) serve(cxt context.Context) {
 			return
 		}
 
+		logCxt.Infof("Accepted from %s", conn.RemoteAddr())
+		tlsConn, ok := conn.(*tls.Conn)
+		if ok {
+			logCxt.Info("TLS connection")
+			err = tlsConn.Handshake()
+			if err != nil {
+				logCxt.WithError(err).Info("TLS handshake error")
```

Comment:
I'm changing this to Panic, same as we do for an error on the Listen or Accept calls.